### DISCUSSION TITLE
Irregular Geometry

### DISCUSSION THREAD

**Discussion by engrmessi** [idaholab/moose#27826] [Q&A Modules: Phase field]
I am struggling with how to create an irregular pit shape for my phase field simulation. Please see attached
![irregular pit](https://github.com/idaholab/moose/assets/125291515/e9f9fff0-5f54-40d1-9729-0aad386b27a7)

**Comment by GiudGiud** [comment]
Hello

If you have the analytical description of this shape use a ParsedFunction and a FUnctionIC
If you have pictures of the irregular shape you can try an ImageFunction and a FunctionIC.

Else you'll want to write a new Function object to describe the shape and use a FunctionIC 

Guillaume
